api/job.go

```diff
@@ -0,0 +1,419 @@
+// Copyright 2023 tsuru authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package api
+
+import (
+	stdContext "context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/tsuru/tsuru/auth"
+	"github.com/tsuru/tsuru/errors"
+	"github.com/tsuru/tsuru/event"
+	tsuruIo "github.com/tsuru/tsuru/io"
+	"github.com/tsuru/tsuru/job"
+	"github.com/tsuru/tsuru/permission"
+	"github.com/tsuru/tsuru/provision"
+	appTypes "github.com/tsuru/tsuru/types/app"
+	jobTypes "github.com/tsuru/tsuru/types/job"
+	permTypes "github.com/tsuru/tsuru/types/permission"
+)
+
+type inputJob struct {
+	TeamOwner   string            `json:"team-owner"`
+	Plan        string            `json:"plan"`
+	Name        string            `json:"name"`
+	Description string            `json:"description"`
+	Pool        string            `json:"pool"`
+	Metadata    appTypes.Metadata `json:"metadata"`
+
+	Schedule  string                 `json:"schedule"`
+	Container jobTypes.ContainerInfo `json:"container"`
+	Trigger   bool                   `json:"trigger"` // Trigger means the client wants to forcefully run a job or a cronjob
```

Comment:
Honestly, I'd change the `Trigger` property name to `Run` or `Execute`. "Trigger" might conflict with some mecanism we can attach with job to trigger it in the future. For example: I want my job be triggered from a webhook. Do you see?